libcontainerd/client_daemon.go

```diff
@@ -114,6 +114,13 @@ type client struct {
 	containers map[string]*container
 }
 
+func (c *client) reconnect() error {
```

Comment:
We probably need to catch connection failures in normal requests and trigger a reconnect, but we aren't handling these cases at all (or ever AFAIK).

Reply:
Those requests would just fail until the connection monitor reconnects normally.